autodiff/reverse/var/var.hpp

```diff
@@ -749,9 +749,9 @@ struct PowExpr : BinaryExpr<T>
     {
         const auto lval = l->val;
         const auto rval = r->val;
-        const auto aux = wprime * val;
-        l->propagate(aux * rval / lval);
-        r->propagate(aux * log(lval));
+        const auto aux = wprime * pow(lval, rval - 1);
+        l->propagate(aux * rval);
+        r->propagate(aux * lval * log(lval));
```

Comment:
Did you check this work? I'm afraid this will produce NaN as well, because of `log(lval)`.

Reply:
Maybe we need something like this:

~~~
const auto auxr = lval == 0.0 ? 0.0 : lval * log(lval);
 r->propagate(aux * auxr);
~~~

Reply:
It passes the failing test I added and works in the rosenbrock example https://github.com/autodiff/autodiff/issues/178#issue-976690033. It's no worse than the existing expressions in the `propagatex` functions on lines 761 and 782, which also take the log of `lval`.

Reply:
It could be that since I was only looking at `pow(x,y)` for constant `y`, the expression for the derivative w.r.t. `y` is not being propagated.